src/flow_solver/flow_solver.cpp

```diff
@@ -337,6 +337,32 @@ void FlowSolver<dim,nstate>::output_restart_files(
 }
 #endif
 
+template <int dim, int nstate>
+void FlowSolver<dim,nstate>::perform_steady_state_mesh_adaptation() const
+{
+    std::unique_ptr<MeshAdaptation<dim,double>> meshadaptation = std::make_unique<MeshAdaptation<dim,double>>(this->dg, &(this->all_param.mesh_adaptation_param));
+    const int total_adaptation_cycles = this->all_param.mesh_adaptation_param.total_mesh_adaptation_cycles;
+    double residual_norm = this->dg->get_residual_l2norm();
+    
+    pcout<<"Running mesh adaptation cycles..."<<std::endl;
+    while (meshadaptation->current_mesh_adaptation_cycle < total_adaptation_cycles)
+    {
+        // Check if steady state solution is being used.
+        if(residual_norm > ode_param.nonlinear_steady_residual_tolerance)
+        {
+            pcout<<"Warning: Mesh adaptation is currently implemented for steady state flows and the current residual norm isn't sufficiently low."
+                 <<" Please check if steady_state() has been called before adapting the mesh."<<std::endl;
```

Comment:
With the current implementation, this is always the case. Maybe update this comment with different instructions.

Reply:
Updated it in the recent commit.